src/pylhe/__init__.py

```diff
@@ -229,38 +232,32 @@ def readNumEvents(file):
 
 
 def visualize(event, outputname):
+    """
+    create a PDF with a visualisation of the LHE event record as a directed graph
+    """
+    
+    # retrieve mapping of PDG ID to particle name as LaTeX string
+    _PDGID2LaTeXNameMap, _ = DirectionalMaps(
+        "PDGID", "LATEXNAME", converters=(int, str)
+    )
+    # draw graph
     g = nx.DiGraph()
     for i, p in enumerate(event.particles):
         g.add_node(i, attr_dict=p.__dict__)
-        name = pypdt.particle(p.id).name
-        greek = [
-            "gamma",
-            "nu",
-            "mu",
-            "tau",
-            "rho",
-            "Xi",
-            "Sigma",
-            "Lambda",
-            "omega",
-            "Omega",
-            "Alpha",
-            "psi",
-            "phi",
-            "pi",
-            "chi",
-        ]
-        for greekname in greek:
-            if greekname in name:
-                name = name.replace(greekname, "\\" + greekname)
-        if "susy-" in name:
-            name = name.replace("susy-", "\\tilde ")
-        g.node[i].update(texlbl="${}$".format(name))
+        try:
+            iid = int(p.id)
+            name = _PDGID2LaTeXNameMap[iid]
+            texlbl = "${}$".format(name)
+        except:
```

Comment:
We're currently failing CI here given `flake8`'s [E722 do not use bare 'except'](https://www.flake8rules.com/rules/E722.html) I'll fix this here as soon as I'm out of my next meeting.